Doc/library/types.rst

```diff
@@ -86,6 +86,34 @@ Dynamic Type Creation
 
    :pep:`560` - Core support for typing module and generic types
 
+.. function:: get_original_bases(cls, /)
+
+    Return the objects in the bases list of the class's definition as they
+    existed prior to any modification by :meth:`~object.__mro_entries__`. This
+    is useful for introspecting :ref:`Generics <user-defined-generics>`.
+
+    Examples::
+
+        from typing import TypeVar, Generic
+
+        T = TypeVar("T")
+        class Foo(Generic[T]): ...
+        class Bar(Foo[int], float): ...
+        class Baz(list[str]): ...
+
+        Foo.__bases__ == (Generic,)
+        get_original_bases(Foo) == (Generic[T],)
+
+        Bar.__bases__ == (Foo, float)
+        get_original_bases(Bar) == (Foo[int], float)
+
+        Baz.__bases__ == (list,)
+        get_original_bases(Baz) == (list[str],)
+
+        get_original_bases(int) == (object,)
```

Comment:
```suggestion
        int.__bases__ == (object,)
        get_original_bases(int) == (object,)
```

Reply:
Why was this marked as "resolved" but not applied? Do you disagree with it? :)

Reply:
Oh sorry I must have misclicked